File: terminal_renderer.py


class TerminalRenderer:
    def __init__(self, gameboard):
        self.gameboard = gameboard

    def clear(self):
        print('\x1b[2J\x1b[H')

    def render(self):
        board = self.gameboard.board
        output = ''
        
        for row in board:
            for spot in row:
                sprite = spot.piece.sprite
                char = sprite[spot.sub_y][spot.sub_x]
                output += char
            output += '\n'

        self.clear()
        print(output)

File: src/game-logic/terminal-renderer.js
class TerminalRenderer {
  constructor (gameboard, domController) {
    this.gameboard = gameboard
    this.domController = domController
  }

  render () {
    let board = this.gameboard.board
    let output = ''

    for (let y = 0; y < board.length; y++) {
      let row = board[y]
      for (let x = 0; x < row.length; x++) {
        let spot = row[x]
        let sprite = spot.piece.sprite
        let char = sprite[spot.subY][spot.subX]
        output += char
      }
      output += '\n'
    }

    this.domController.updateTerminal(output)
  }
}

export default TerminalRenderer